dev-client/src/components/SoilPropertiesDataTable.tsx

```diff
@@ -0,0 +1,112 @@
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see https://www.gnu.org/licenses/.
+ */
+import {useTranslation} from 'react-i18next';
+import {
+  Box,
+  Row,
+  Text,
+} from 'terraso-mobile-client/components/NativeBaseAdapters';
+import {NBDimensionValue} from 'terraso-mobile-client/components/util/nativeBaseAdapters';
+
+type DataTableHeaderProps = {
+  width: NBDimensionValue;
+  text: string;
+};
+const DataTableHeader = ({width, text}: DataTableHeaderProps) => {
+  return (
+    <Text
+      variant="table-header"
+      width={width}
+      paddingHorizontal={'sm'}
+      pb={'sm'}
+      alignSelf={'flex-end'}>
+      {text}
+    </Text>
+  );
+};
+
+type DataTableCellProps = {
+  text: string;
+  index: number;
+  width: NBDimensionValue;
+};
+const DataTableCell = ({text, index, width}: DataTableCellProps) => {
+  return (
+    <Box
+      flex={1}
+      key={index}
+      width={width}
+      borderRightWidth="1px"
+      borderBottomWidth="1px"
+      paddingHorizontal="8px"
+      paddingVertical="4px"
+      justifyContent="center">
+      <Text>{text}</Text>
+    </Box>
+  );
+};
+
+type Props = {
+  rows: [string, string, string, string][];
+} & React.ComponentProps<typeof Box>;
+
+export const SoilPropertiesDataTable = ({rows, ...containerProps}: Props) => {
+  const {t} = useTranslation();
+
+  const columnWidthDepth: NBDimensionValue = '80px';
+  const columnWidthTexture: NBDimensionValue = '100px';
+  const columnWidthColor: NBDimensionValue = '100px';
+  const columnWidthRockFragment: NBDimensionValue = '80px';
+
+  return (
+    <Box {...containerProps}>
+      <Row justifyContent="flex-start">
+        <DataTableHeader
+          width={columnWidthDepth}
+          text={t('site.soil_id.site_data.soil_properties.depth')}
+        />
+        <DataTableHeader
+          width={columnWidthTexture}
+          text={t('site.soil_id.site_data.soil_properties.texture')}
+        />
+        <DataTableHeader
+          width={columnWidthColor}
+          text={t('site.soil_id.site_data.soil_properties.color')}
+        />
+        <DataTableHeader
+          width={columnWidthRockFragment}
+          text={t('site.soil_id.site_data.soil_properties.rock_fragment')}
+        />
+      </Row>
+
+      <Box borderTopWidth={'1px'} borderLeftWidth={'1px'}>
+        {rows.map((row: (typeof rows)[number], i: number) => (
+          <Row justifyContent="flex-start" key={i}>
```

Comment:
question: do we need the key? I haven't seen it elsewhere in react native code.

Reply:
Oo as a result of this comment, I just learned that while you should use a key on mapped components, it is considered an anti-pattern to use the index alone as the key.
(https://robinpokorny.medium.com/index-as-a-key-is-an-anti-pattern-e0349aece318)

Reply:
Yeah, we do that a bunch in regular React. I hadn't seen it elsewhere in the bo app, which is why I asked. Do we get the warning without it?

Reply:
Yeah, I tried removing the key and the warning does show up.